res/controllers/Reloop-Mixage.scripts.js

```diff
@@ -0,0 +1,812 @@
+Mixage.wheelTurn = function(_channel, _control, value, _status, group) {
+    const deckNr = script.deckFromGroup(group);
+    const diff = value - 64; // 0x40 (64) centered control
+    if (Mixage.adjustLoop[group]) {  // loop adjustment
+        // triple the adjustment rate if the top of the jogwheel is being touched
+        const factor = Mixage.wheelTouched[group] ? 100 : 33;
+        if (Mixage.adjustLoopIn[group]) {
+            const newStartPosition = engine.getValue(group, "loop_start_position") + (diff * factor);
+            if (newStartPosition < engine.getValue(group, "loop_end_position")) {
+                engine.setValue(group, "loop_start_position", newStartPosition);
+            }
+        }
+        if (Mixage.adjustLoopOut[group]) {
+            const newEndPosition = engine.getValue(group, "loop_end_position") + (diff * factor);
+            if (newEndPosition > engine.getValue(group, "loop_start_position")) {
+                engine.setValue(group, "loop_end_position", newEndPosition);
+            }
+        }
+    } else if (Mixage.scratchByWheelTouch || Mixage.scratchToggleState[group] || Mixage.scrollToggleState[group]) {
+        if (Mixage.scrollToggleState[group]) { // scroll deck
+            // triple the scroll rate if the top of the jogwheel is being touched
+            const speedFactor = Mixage.wheelTouched[group] ? 0.00020 : 0.000066;
+            const currentPosition = engine.getValue(group, "playposition");
+            engine.setValue(group, "playposition", currentPosition + speedFactor * diff * Mixage.jogWheelScrollSpeed);
+        } else if (Mixage.wheelTouched[group]) {
+            engine.scratchTick(deckNr, diff); // scratch deck
+        } else {
+            engine.setValue(group, "jog", diff); // pitch bend deck
+        }
+    }
+};
+
+// stop or start loop adjustment mode
+Mixage.handleBeatLoopPress = function(_channel, _control, value, _status, group) {
+    if (Mixage.adjustLoop[group] && value === DOWN) {
+        Mixage.stopLoopAdjust(group);
+    } else if (value === DOWN && engine.getValue(group, "loop_start_position") !== -1 && engine.getValue(group, "loop_end_position") !== -1) {
+        Mixage.adjustLoop[group] = true;
+        Mixage.startLoopAdjust(group);
+    }
+};
+
+// move the track or an active loop "beatjump_size" number of beats
+Mixage.handleBeatMove = function(_channel, _control, value, _status, group) {
+    const beatjumpSize = (value - 64) * engine.getValue(group, "beatjump_size");
+    engine.setValue(group, "beatjump", beatjumpSize);
+};
+
+// clears a loop on a short press, set internal variable to true to adjust "beatjump_size"
+Mixage.handleLoopLengthPress = function(_channel, _control, value, _status, group) {
+    if (value === DOWN) {
+        Mixage.loopLengthPressed[group] = true;
+        Mixage.loopLengthPressTimer = engine.beginTimer(400, function() {
+            Mixage.loopLengthPressTimer = 0;
+        }, true);
```

Comment:
have a timer per group, like the longLengthPressed bool?

Reply:
I didn't think this was necessary as under normal use it is unlikely a user would clear the loop on both decks at the same time for instance. I added a timer per group now just to be safe